### PR TITLE
Fix mutliple listeners, remove whitespace, and dead code

### PR DESCRIPTION
**Pull request by michaelkm**
@rrgayhart Will you check this out?
Also, @tgraham777, you would just love this.

What's this PR do?

* Adds wrapper div to Layout to eliminate multiple listeners.
* Removes listeners from getIdeas and adds them in the initialize. This fixed the multiple listeners adding up issue.
* Removed whitespace and cleaned up formatting according to JSHint.
* Removed dead console.logs from debugging.

Where should the reviewer start?
Start with ideaBox.getIdeas() and follow the listeners.

How should this be manually tested?
Visiting the homepage and adding and updating the ideas on the page.

Any background context you want to provide?
Before the refactor, there were multiple event listeners being attached every time an idea was edited or added. This slowed down the performance of the app. That doesn't happen anymore. 

Screenshots (if appropriate)

What gif best describes this PR or how it makes you feel?
I'm not listening!

### CODE REVIEW

File: app/assets/javascripts/welcome.js [hunk 1]
```diff
@@ -1,215 +1,209 @@
...
-      var $idea = $( this ).closest( '.idea' );
-      console.log("removeIdea");
+  removeIdea: function() {
+    $('#wrapper').on('click', '.glyphicon-remove', function() {
+      var $idea = $(this).closest('.idea');
       $.ajax({
-        url: '/api/v1/ideas/' + $idea.attr( 'id' ),
-        type: 'POST',
+        url: '/api/v1/ideas/' + $idea.attr('id'),
+        type: 'DELETE',
```
Comment (rrgayhart) [inline app/assets/javascripts/welcome.js]: Nice!

File: app/assets/javascripts/welcome.js [hunk 2]
```diff
@@ -1,215 +1,209 @@
...
-    if(string.length >= 100) {
+  trimBody: function(string) {
+    if (string.length >= 100) {
       var cut = string.lastIndexOf(' ', 100);
       string =  string.substring(0, cut) + '...';
     }
+
     return string;
-  }
+  },
```
Comment (rrgayhart) [inline app/assets/javascripts/welcome.js]: Do you need this feller?

File: app/assets/javascripts/welcome.js [hunk 3]
```diff
@@ -1,215 +1,209 @@
...
+      url: '/api/v1/ideas/' + $idea.attr('id'),
+      data: { idea: { quality: quality } },
+      success: function() {
         ideaBox.getIdeas();
       },
-      error: function(err){
-        console.log("Failed!:" + err);
-      }
+
+      error: function(err) {
```
Comment (rrgayhart) [inline app/assets/javascripts/welcome.js]: I believe that you don't actually need an error statement here, if you aren't doing anything within the block.

**Comment by rrgayhart** [comment]
LGTM - just a few small comments!

**Comment by rrgayhart** [comment]
Fixes #3

File: app/assets/javascripts/welcome.js (hunk 1 above)
Comment (tgraham777) [inline app/assets/javascripts/welcome.js]: Good idea extracting to the init function!

File: app/assets/javascripts/welcome.js [hunk 4]
```diff
@@ -1,215 +1,209 @@
 var ideaBox = {
-  init: function(){
+  init: function() {
     ideaBox.getIdeas();
     ideaBox.createIdea();
+    ideaBox.truncate();
```
Comment (tgraham777) [inline app/assets/javascripts/welcome.js]: Good idea extracting these to the init function!

File: app/assets/javascripts/welcome.js (hunk 1 above)
Comment (tgraham777) [inline app/assets/javascripts/welcome.js]: Makes sense!

**Comment by tgraham777** [comment]
The way you solved your multiple listener problem helped me understand why I keep running into the same issue. Good work on this!

**Pull request merged**